bin/radical-pilot-component

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+__copyright__ = "Copyright 2014-2019, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+import sys
+import time
+
+import setproctitle  as spt
+import radical.utils as ru
+import radical.pilot as rp
+
+dh = ru.DebugHelper()
+
+
+# ------------------------------------------------------------------------------
+#
+def main(cfg):
+    '''
+    This thin wrapper starts a RCT component It expects a single argument:
+    a config to use for the component's configuration.  The config must contain:
+
+      - uid : UID of component instance (unique to the hosting session)
+      - name: name of the component
+      - kind: type of component
+
+    If the config contains a `heartbeat` section, that section must be formatted
+    as follows:
+
+        {
+          'from'    : 'uid',
+          'pub'     : 'addr_pub',
+          'sub'     : 'addr_sub',
+          'interval': <float>,
+          'timeout' : <float>
+        }
+
+    If that section exists, the process will daemonize and heartbeats are used
+    to manage the bridge lifetime: the lifetime of this bridge is then dependent
+    on receiving heartbeats from the given `uid`: after `timeout` seconds of no
+    heartbeats arriving, the bridge will terminate.  The bridge itself will
+    publish heartbeats every `interval` seconds on the heartbeat channel under
+    its own uid.
+
+    If the heartbeat section is not present in the config file, the components
+    lifetime is expected to be explicitly managed, i.e., that this wrapper
+    process hosting the bridge is terminated externally.
+
+    The config file may contain other entries which are passed to the component
+    and are interpreted by the component implementation.
+    '''
+
+    # basic setup: uid, logger, process name
+    uid  = cfg.uid
+    path = cfg.path
+    prof = ru.Profiler(name=uid, ns='radical.pilot', path=path)
+
+    try:
+        prof.prof('comp_start', uid=uid)
+        wrapped_main(cfg, uid, path)
+    finally:
+        prof.prof('comp_stop', uid=uid)
+
+
+def wrapped_main(cfg, uid, path):
+
+    log = ru.Logger(name=uid, ns='radical.pilot', path=path)
+
+    spt.setproctitle('rp.%s' % uid)
+
+    # start a non-primary session
+    session = rp.Session(cfg=cfg, _primary=False)
+
+    # create the component and begin to work
+    comp  = rp.utils.Component.create(cfg, session)
+    comp.start()
+
+    # component runs - send heartbeats so that cmgr knows about it
+    hb_pub = ru.zmq.Publisher ('heartbeat', cfg.heartbeat.addr_pub)  #, log=log)
```

Comment:
We wrap main only when demonizing right? Maybe adding it in the docstring of the function?

Reply:
`main(cfg)` is running in the daemonized process.  I only introduced `wrapped_main` to handle the nesting (indentation).  In general, the `except` should not be too far away from the `try` to keep it coherent and readable, and I found the code which is now in `wrapped_main` too long for that.

That `main` is daemonized should be visible from 
```
ru.daemonize(main=main, args=[cfg], stdout='%s.out' % path,
                                        stderr='%s.err' % path)
```
and from the `ru.daemonize` documentation.  Let me know if you think that's insufficient!